[RM Core Appendix A] Added VNF Monitoring UseCase in Appendix-A [Fixes #387]

ashokkumarprajapati opened the pull request:
Added VNF Monitoring UseCase in Appendix-A [Fixes #387]

kedmison commented:
For this to be a use case, it should not talk about 'shoulds' in terms of requirements, but more about the 'hows'.  i.e. how the measurements and capabilities identified in earlier parts of the document come together to enable the sorts of VNF network infrastructure issue identification envisioned.  If the measurements and capabilities don't exist, then we should discuss whether to identify and add them to the right sections.

petorre commented:
@ashokkumarprajapati Could you please write few line summary of "UseCase for Exposed Monitoring capabilities" or "VNF Monitoring UseCase" just to be clear on who is monitoring whom for which expected outcome?

rabiabdel commented:
@ashokkumarprajapati would you like to continue this PR. there has been no activity on it for a while. can you please address comments above.

ashokkumarprajapati commented:
@petorre Done. Updated heading for better understanding.
@kedmison : Updated usecase with example like vTAP. However the idea behind this was to explain 'why' instead 'how'. 'How' would be part of RA/RI.

petorre commented:
@ashokkumarprajapati Thanks for the description. This needs to be also functionality in RAs like in [RA1\Ch7.4](https://github.com/cntt-n/CNTT/blob/master/doc/ref_arch/openstack/chapters/chapter07.md#7.4) and/or [RA2\Ch7](https://github.com/cntt-n/CNTT/blob/master/doc/ref_arch/kubernetes/chapters/chapter07.md) . While writing it there, I recommend being very specific that any such platform functionality/module comes from platform provider and not from tenant, and carefully deciding optional vs mandatory as it can get complex with integration+operational impact.
For now we can keep the text here while writing Appendix towards VNFs.
Please comments from @ashokkumarprajapati , @rabiabdel , @markshostak and team?

ashokkumarprajapati commented on doc/ref_model/chapters/appendix-a.md:
```diff
@@ -26,3 +27,15 @@ The goal is not to be prescriptive on how to re-architect existing or architect
+
+<a name="A.5"></a>
+## A.5 Miscellaneous
+### A.5.1 VNF Monitoring Capabilities - UseCase.
```
I agree that current description is more towards VNF network troubleshooting area. Updated the title accordingly.
@petorre : Updated description to specifically mention that it is NFVI platform is responsible for exposing it.

markshostak replied:
Ashok, This is good. It would be great if you could put some more narrative around using the feature, the scenario, etc. 

To that point, you mentioned leveraging a vTAP, but you also mentioned the NFVI would have to expose the capability. I can see the vTAP approach if the monitoring is intra-VNF. However, if the monitoring is being performed inside the monitored VNF itself (intra-VNF), an Infra-based monitoring capability is not required and the solution should be implemented at the discretion of the VNF designer, not CNTT. 

If the monitoring is inter-VNF, which I suspect is more likely, I would expect the mirrored traffic to be exposed to the monitoring VNF on a typical virtual interface, bubbling up from the Infra and ingressing the relevant VNF, the same as standard pt-to-pt traffic. What network topology/scenario were you envisioning with w/ the vTAP implementation? Thanks, -Mark

ashokkumarprajapati replied:
@markshostak : Thanks. I have explained now how NFVI must provide capability to expose mirrored traffic on new virtual interface so that monitoring VNF can use it.
I currently do not see any other monitoring topology other then pt-to-pt as of now.

markshostak replied:
Thank you. I'll take the liberty and Resolve this thread for Kelvin. @kedmison if you disagree, ping me and let's discuss. Thanks, -Mark

rabiabdel commented:
@markshostak can you please review

markshostak commented:
@ashokkumarprajapati Hi Ashok, I'm not seeing commits implementing the changes you reference above, in the PR's Commits tab. It only shows 1 commit on the commits tab, but there are mentions of commits pushed, farther down. Usually they show up on that tab. @rabi-abdel Maybe Rabi can explain why GH is eating the updates. :-)

rabi-abdel commented:
> @ashokkumarprajapati Hi Ashok, I'm not seeing commits implementing the changes you reference above, in the PR's Commits tab. It only shows 1 commit on the commits tab, but there are mentions of commits pushed, farther down. Usually they show up on that tab. @rabi-abdel Maybe Rabi can explain why GH is eating the updates. :-)

@Markshostak @ashokkumarprajapati Github is not eating the updates, there is only one commit to this PR.

deebhatia commented:
@markshostak @rabi-abdel This is due to amending the same commit to include requested changes resulting in a `force push` which replaces the older commit instead of adding a new one. To see what was updated in a force push click the link embedded in `force-push` text.
<img width="670" alt="Screenshot 2019-11-26 at 3 07 17 PM" src="https://user-images.githubusercontent.com/5617891/69617878-eac57200-105e-11ea-9a48-d8c4d2227a40.png">

markshostak commented:
Thanks Deepanshu. @ashokkumarprajapati Hi Ashok, Going forward, please just add additional commits when editing. Updating the existing commit makes it very difficult to see what's changed. Thanks, -Mark

rabi-abdel commented:
@markshostak contributing policy updated.  @kedmison once you are happy with the updated by @ashokkumarprajapati please resolve your comments and approve.

rabi-abdel commented:
@ashokkumarprajapati @markshostak I have been looking at this for sometime, if vTAP features is needed, it needs to go to RA1 Ops and RA2 Ops. in terms of capabilities, it needs to be reflected in RM Ch04. not sure why this content is being proposed for RM Appendix. I am inclined to "return" this. do you agree @markshostak ?

ashokkumarprajapati commented:
> @ashokkumarprajapati @markshostak I have been looking at this for sometime, if vTAP features is needed, it needs to go to RA1 Ops and RA2 Ops. in terms of capabilities, it needs to be reflected in RM Ch04. not sure why this content is being proposed for RM Appendix. I am inclined to "return" this. do you agree @markshostak ?

@rabi-abdel : Well, feature is supposed to be for sure part of RA1, RA2. However original idea to add it here was to explain what are Monitoring capabilities and why theses are needed. It was aimed to clear confusion between NFVI monitoring v/s Capabilities from NFVI to monitor VNFs.

markshostak commented:
> 
> 
> > @ashokkumarprajapati @markshostak I have been looking at this for sometime, if vTAP features is needed, it needs to go to RA1 Ops and RA2 Ops. in terms of capabilities, it needs to be reflected in RM Ch04. not sure why this content is being proposed for RM Appendix. I am inclined to "return" this. do you agree @markshostak ?
> 
> @rabi-abdel : Well, feature is supposed to be for sure part of RA1, RA2. However original idea to add it here was to explain what are Monitoring capabilities and why theses are needed. It was aimed to clear confusion between NFVI monitoring v/s Capabilities from NFVI to monitor VNFs.

I would have to agree. We're trying to capture a use case; how to implement the capability is a separate discussion. The capability is reflected in RM Ch04. My understanding is, this content was proposed for the appendix so it could provide a story around when, why and how to use the monitoring feature.

As far as next steps, I'd still like to see some more narrative in the use case. There are 3 bullets describing applications for monitoring, but they're at the 500,000m view. How would one use this feature to help resolve performance issues? What type of anomaly and forensics and how does monitoring play into the picture?

I'd suggest ditching the vTAP reference, and any specific implementation suggestions. Once the feature has been designed or at least roughed out, a few comments giving a summary of the mechanics could potentially be appended to the description, but adding it now is veering off course from the objective. For this PR I'd simply focus on describing how someone would derive value from the monitoring feature. Thanks, -Mark

ashokkumarprajapati commented:
> > > @ashokkumarprajapati @markshostak I have been looking at this for sometime, if vTAP features is needed, it needs to go to RA1 Ops and RA2 Ops. in terms of capabilities, it needs to be reflected in RM Ch04. not sure why this content is being proposed for RM Appendix. I am inclined to "return" this. do you agree @markshostak ?
> > 
> > 
> > @rabi-abdel : Well, feature is supposed to be for sure part of RA1, RA2. However original idea to add it here was to explain what are Monitoring capabilities and why theses are needed. It was aimed to clear confusion between NFVI monitoring v/s Capabilities from NFVI to monitor VNFs.
> 
> I would have to agree. We're trying to capture a use case; how to implement the capability is a separate discussion. The capability is reflected in RM Ch04. My understanding is, this content was proposed for the appendix so it could provide a story around when, why and how to use the monitoring feature.
> 
> As far as next steps, I'd still like to see some more narrative in the use case. There are 3 bullets describing applications for monitoring, but they're at the 500,000m view. How would one use this feature to help resolve performance issues? What type of anomaly and forensics and how does monitoring play into the picture?
> 
> I'd suggest ditching the vTAP reference, and any specific implementation suggestions. Once the feature has been designed or at least roughed out, a few comments giving a summary of the mechanics could potentially be appended to the description, but adding it now is veering off course from the objective. For this PR I'd simply focus on describing how someone would derive value from the monitoring feature. Thanks, -Mark

@markshostak : I have now removed mention to vTap and also explained that NFVI needs to ensure mirrored traffic is available on new virtual interface for monitoring VNF.

rabiabdel commented:
Thanks @ashokkumarprajapati @markshostak . let us get this out of the way then. it will be OK to expect future PRs to fill in the gaps if any.

ashokkumarprajapati commented:
@rabi-abdel : Not sure why its failing check. Can you help on this plz.
